zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/TenantDeletedApplier.java

```diff
@@ -47,7 +47,11 @@ private void handleUserEntities(
       final Map<EntityType, List<Long>> entities, final String tenantId) {
     final List<Long> userEntities = entities.get(EntityType.USER);
     if (userEntities != null) {
-      userEntities.forEach(userKey -> userState.removeTenant(userKey, tenantId));
+      userEntities.forEach(
+          userKey ->
+              userState
+                  .getUser(userKey)
+                  .ifPresent(user -> userState.removeTenant(user.getUsername(), tenantId)));
     }
   }
```

Comment:
🔧 We can remove this code completely. In the `TenantDeleteProcessor` [here](https://github.com/camunda/camunda/blob/a406f791bb017b34f4d20f0c5304f267efb814ac/zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/tenant/TenantDeleteProcessor.java#L127-L144), we emit `TenantIntent.ENTITY_REMOVED` events for each member entity, so the `TenantEntityRemovedApplier` is already invoked to remove these entries from the `userState`.